From 00e9823c1af38497393c03813a955e0f8e1f73c9 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Tue, 5 May 2026 21:39:22 +0200
Subject: [PATCH 93/94] net: dpaa2-mac: close mac on remove path

When standalone mac driver is removed, it should first disconnect mac,
then close it.

The close call was missing, add it.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c b/drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c
index 01396e761f32..8f04d88e67c9 100644
--- a/drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c
+++ b/drivers/net/ethernet/freescale/dpaa2/dpaa2-mac.c
@@ -1205,10 +1205,9 @@ static int dpaa2_mac_remove(struct fsl_mc_device *mc_dev)
 
 	dpaa2_mac_teardown_irqs(mc_dev);
 
-	if (dpaa2_mac_is_type_phy(priv)) {
-		dpaa2_mac_close(priv);
+	if (dpaa2_mac_is_type_phy(priv))
 		dpaa2_mac_disconnect(priv);
-	}
+	dpaa2_mac_close(priv);
 
 	fsl_mc_portal_free(mc_dev->mc_io);
 	dev_set_drvdata(dev, NULL);
-- 
2.51.0

